onnxruntime/core/framework/utils.cc

```diff
@@ -663,27 +844,77 @@ common::Status ExecutePartialGraph(const SessionState& session_state, FeedsFetch
       p_fetches = &device_fetches;
     }
 
-    ORT_RETURN_IF_ERROR(executor.Execute(session_state,
-                                         feeds_fetches_info.feeds_mlvalue_idxs, p_feeds,
-                                         feeds_fetches_info.fetches_mlvalue_idxs, *p_fetches, {},
-                                         logger));
+    ORT_RETURN_IF_ERROR(PartialExecuteThePlan(session_state,
+                                              feeds_fetches_info.feeds_mlvalue_idxs, p_feeds,
+                                              feeds_fetches_info.fetches_mlvalue_idxs, *p_fetches, {},
+                                              logger,
+                                              device_stream_collection,
+                                              terminate_flag,
+                                              // single thread mode
+                                              single_thread_mode,
+                                              state,
+                                              cache));
+
+    InlinedVector<Stream*> fetches_streams;
+    fetches_streams.reserve(feeds_fetches_info.fetches_mlvalue_idxs.size());
+    auto& value_to_stream_map = execution_plan->value_to_stream_map;
+    for (auto fetch_idx : feeds_fetches_info.fetches_mlvalue_idxs) {
+      auto it = value_to_stream_map.find(fetch_idx);
+      if (it != value_to_stream_map.end()) {
+        fetches_streams.push_back(device_stream_collection.GetStream(it->second));
+      } else {
+        // for subgraph, it is possible the graph is empty,
+        // the fetches are come from parent graph.
+        fetches_streams.push_back(parent_stream);
+      }
+    }
 
     if (device_copy_checks.output_copy_needed == DeviceCopyCheck::Copy) {
-      ORT_RETURN_IF_ERROR(CopyOutputsAcrossDevices(session_state, *p_fetches, fetches, fetch_copy_info));
+      ORT_RETURN_IF_ERROR(CopyOutputsAcrossDevices(session_state, *p_fetches, fetches, fetch_copy_info, fetches_streams));
     }
+    // training don't want to flush the stream
   }
 
   return Status::OK();
 }
+
+common::Status ExecutePartialGraph(const SessionState& session_state, FeedsFetchesManager& feeds_fetches_manager,
+                                   gsl::span<const OrtValue> feeds, std::vector<OrtValue>& fetches,
+                                   const logging::Logger& logger, PartialGraphExecutionState& state,
+                                   const OrtValueCachePtr& cache, const bool& terminate_flag,
+                                   // TODO: handle the partial_graph_index
```

Comment:
>// TODO: handle the partial_graph_index [](http://example.com/codeflow?start=35&length=39)

To give more context here, partial_graph_index is used for memory profiler. 
fw run the function, started the memory profiler. 
bw run the function a second time, ended the memory profiler.

Reply:
offline synced, @randy will fix the memory profiling in session scope. for the partial run, we will fix it later.